server/services/offender/index.js

```diff
@@ -116,6 +116,25 @@ const createOffenderService = (
     }
   }
 
+  async function getVisitBalances({ prisonerId }) {
+    try {
+      logger.info(`OffenderService (getVisitBalances) - User: ${prisonerId}`);
+
+      if (!prisonerId) {
+        throw new Error('No prisonerId passed');
+      }
```

Comment:
I know this is based on existing patterns but not a fan of these preconditions , especially considering it gets caught by the surrounding try.

Not sure what you think, I guess I would prefer we didn't even call the service layer if we don't have a prisoner ID - I imagine it never actually happens due to the way we only add personalisation if we have a logged in user  so could just get rid?

Also not sure why we don't just log the stack trace in the catch.

Could we ditch the debug of stack trace if we capture using sentry?

Reply:
Good call; this check is pointless. Because I've destructed, the error would already have blown this function up.